docs/upgrade-guides/node-to-express.mdx

```diff
@@ -0,0 +1,197 @@
+  console.log(`Server is running on http://localhost:${port}`)
+})
+```
+
+## `requireAuth()` usage changes
+
+In the Node SDK, `requireAuth()` wrapped route handlers. With the Express SDK, you must instead pass `requireAuth` as a middleware function before your handler instead of wrapping your handler.
+
+```ts {{ del: [1, 4], ins: [2, 5] }}
+import { requireAuth } from '@clerk/clerk-sdk-node'
+import { requireAuth } from '@clerk/express'
+
+app.get('/', requireAuth(handler))
+app.get('/', requireAuth(), handler)
+```
+
+## Migrate from `ClerkExpressWithAuth`
+
+To replace your existing Node SDK middleware [`ClerkExpressWithAuth()`](/docs/backend-requests/handling/nodejs), which allows requests to proceed even when users are not authenticated, you need to write your own middleware that uses [`getAuth()`](/docs/references/express/overview#get-auth) to retrieve the auth state and then return an empty object if the user is not authenticated.
+
+```ts {{ del: [1, [19, 21]], ins: [2, 8, 9, [11, 16], [22, 24]] }}
+import { ClerkExpressWithAuth } from '@clerk/clerk-sdk-node'
+import { clerkMiddleware, getAuth } from '@clerk/express'
+import express from 'express'
+
+const app = express()
+const port = 3000
+
+// clerkMiddleware is required to be set in the middleware chain before getAuth is used
+app.use(clerkMiddleware())
+
+const withAuth = (req, res, next) => {
+  const { userId } = getAuth(req)
+  // If the user is not authenticated, return an empty object
+  req.auth = userId ? auth : {}
+  next()
+}
+
+// Define a protected route
+app.get('/protected-route', ClerkExpressWithAuth(), (req, res) => {
+  res.json(req.auth)
+})
+app.get('/protected', withAuth, (req, res) => {
+  res.json(req.auth)
+})
+
+app.listen(port, () => {
+  console.log(`Server is running on http://localhost:${port}`)
+})
+```
+
+## Migrate from `ClerkExpressRequireAuth`
+
+To enforce strict authentication, replace your existing Node SDK strict middleware, [`ClerkExpressRequireAuth`](/docs/backend-requests/handling/nodejs), with the new Express SDK middleware, [`requireAuth`](/docs/references/express/overview#require-auth).
+
+`requireAuth` is a middleware function that you can use to protect routes in your Express.js application. This function checks if the user is authenticated, and redirects to the configured sign in URl if they are not.
+
+```ts {{ del: [1, [9, 11]], ins: [2, 3, [12, 14]] }}
+import { ClerkExpressRequireAuth, RequireAuthProp, StrictAuthProp } from '@clerk/clerk-sdk-node'
+import { clerkMiddleware, requireAuth } from '@clerk/express'
+import express from 'express'
+
+const app = express()
+const port = 3000
+
+// Define a protected route
+app.get('/protected', ClerkExpressRequireAuth(), (req, res) => {
+  res.send('This is a protected route')
+})
+app.get('/protected', requireAuth(), (req, res) => {
+  res.send('This is a protected route')
+})
+
+app.listen(port, () => {
+  console.log(`Server is running on http://localhost:${port}`)
+})
+```
+
```

Comment:
````suggestion

If you'd like to retain the previous behavior of `ClerkExpressRequireAuth` (emitting an error rather than redirecting), you can accomplish this as such:

```
const legacyRequireAuth = (req, res, next) => if (!req.auth) throw new Error('Unauthenticated')
app.get('/', legacyRequireAuth, handler)
```

You also want to make sure to catch and handle the error using an [error middleware](https://expressjs.com/en/guide/error-handling.html) below.
````